front-end/src/views/function/components/AddFunc.vue

```diff
@@ -0,0 +1,242 @@
+             label-width="80px"
+             label-position="top"
+             style="width: 90%;margin: 25px"
+             size="small"
+             :inline="true" >
+      <el-form-item label="FunctionName:"
+                    style="width: 100%"
+                    prop="functionName">
+        <el-input v-model="form.functionName"
+                  placeholder="请输入FunctionName"
+                  clearable>
+        </el-input>
+      </el-form-item>
+      <a style="color: red;font-size: 14px;">
+        *&nbsp;
+      </a>
+      <a style="font-size: 14px;">
+        Inputs:
+      </a>
+      <el-form-item style="width: 100%;margin-top: 5px"
+                    v-for="(value,index) in form.inputs"
+                    :prop="'inputs.'+index"
+                    :key="index"
+                    :rules="{
+                      required: true, message: '请输入Input', trigger: 'blur'
+                    }">
+        <el-input style="width: 95%"
+                  placeholder="请输入Input"
+                  v-model="form.inputs[index]"
+                  clearable>
+        </el-input>
+        <i style="float: right;font-size: 30px;width: 5%"
+           v-show="form.inputs.length!==1"
+           class="el-icon-circle-close"
+           @click="deleteInput(index)">
+        </i>
+      </el-form-item>
+      <el-button style="width: 100%;font-size: 20px;margin-top: 5px"
+                 size="mini"
+                 class="el-icon-circle-plus-outline"
+                 @click="addInput">
+      </el-button>
+      <el-form-item label="Output:"
+                    style="width: 100%"
+                    prop="output">
+        <el-input v-model="form.output"
+                  placeholder="请输入Output"
+                  clearable>
+        </el-input>
+      </el-form-item>
+      <el-form-item label="Log-topic:"
+                    style="width: 100%"
+                    prop="logTopic">
+        <el-input v-model="form.logTopic"
+                  placeholder="请输入Log-topic"
+                  clearable>
+        </el-input>
+      </el-form-item>
+      <el-form-item label="ClassName:"
+                    style="width: 66%"
+                    prop="className">
+        <el-input v-model="form.className"
+                  placeholder="请输入ClassName"
+                  clearable>
+        </el-input>
+      </el-form-item>
+      <el-form-item label="Runtime:"
+                    style="width: 30%"
+                    prop="runtime">
+        <el-select v-model="form.runtime">
+          <el-option label="JAVA"
+                     value="JAVA">
+          </el-option>
+          <el-option label="GO"
+                     value="GO">
+          </el-option>
+        </el-select>
+      </el-form-item>
+      <el-form-item class="upload"
+                    label="File:"
+                    prop="file"
+                    style="width: 100%">
+        <el-upload
+          :limit="1"
+          drag
+          ref="file"
```

Comment:
Codacy has a fix for the issue: [Attribute "ref" should go before "drag".](https://app.codacy.com/gh/FunctionStream/function-stream/pullRequest?prid=7834759)
```suggestion
          drag
```